packages/legend-query/src/components/QueryBuilderProjectionPanel.tsx

```diff
@@ -494,31 +515,64 @@ export const QueryBuilderProjectionPanel = observer(
       (columnState) => columnState.isBeingDragged,
     );
     const handleDrop = useCallback(
-      (item: QueryBuilderExplorerTreeDragSource): void =>
-        projectionState.addColumn(
-          new QueryBuilderSimpleProjectionColumnState(
-            projectionState,
-            buildPropertyExpressionFromExplorerTreeNodeData(
-              queryBuilderState.explorerState.nonNullableTreeData,
-              item.node,
-              projectionState.queryBuilderState.graphManagerState.graph,
+      (
```

Comment:
I dont think there is a difference between DnD into derivation and creating a new derivation. 
We also don't force users to compile during derivation so there are no special checks that run that wouldnt run with this. This has the same risk as doing it with an already defined derivation.

Reply:
I still don't think this feature makes sense unfortunately. We're basically saying we let you create a projection from a function whereas the simple path we want to shape users into doing is to create a derivation and to be able to DnD function there to save typings

Now, I understand the exploratory aspect of this where it goes like, people are curious and playing with the UI then they discover that the functions are dragable, and they are droppable in the projection panel.

But now, consider the consistency aspect of this behavior, when we have filter, how do you plan to support this? You add an adhoc condition?

Reply:
After discussing with @MauricioUyaguari I think I can _bend_ on this one 😄

But as said, my biggest concerns are:
1. Adding a feature like this is nice, but adding new `easter-egg` features like this costs maintenance (both code and docs)
2. We have to be careful of the implication this has to the UX, to me derivation is a dangerous thing to use (could be me being too _shy_ because our support for derivation at this point is too basic), so in the course of exploring the panel, people can accidentally add a derivation in projection and they would wonder what is this thing.
3. To point (1), we now bear the burden of maintaining the consistency in UX, now when we support derivation better in filter, aggregation, etc. we need to make sure this works. I will add more infos to #627